### ISSUE TITLE
ignores .cz-config.js unless there is a package.json

### ISSUE THREAD

**Issue by zayd62** [KnisterPeter/vscode-commitizen#199]
When you create a ```.cz-config.js```, the plugin seems to ignore the config file unless there is a ```package.json``` file pointing to the ```.cz-config.js```.

My ```package.json```:

```
{
  "config": {
    "cz-customizable": {
      "config": "./.cz-config.js"
    }
  }
}
```

My ```.cz-config.js```: 
```
module.exports = {
  types: [
    {
      value: 'feat',
      name: 'A new feature'
    },
    {
      value: 'fix',
      name: 'A bug fix'
    },
    {
      value: 'docs',
      name: 'Documentation only changes'
    },
    {
      value: 'style',
      name: 'Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)',
    },
    {
      value: 'refactor',
      name: 'A code change that neither fixes a bug nor adds a feature',
    },
    {
      value: 'perf',
      name: 'A code change that improves performance',
    },
    {
      value: 'test',
      name: 'Adding missing tests'
    },
    {
      value: 'chore',
      name: 'Changes to the build process or auxiliary tools and libraries such as documentation generation',
    },
    {
      value: 'revert',
      name: 'Revert to a commit'
    },
    {
      value: 'WIP',
      name: 'Work in progress'
    },
  ],

  scopes: [
    { name: 'square' },
    { name: 'triangle' },
    { name: 'circle' }

  ],

  allowTicketNumber: false,
  isTicketNumberRequired: false,
  ticketNumberPrefix: 'TICKET-',
  ticketNumberRegExp: '\\d{1,5}',

  // it needs to match the value for field type. Eg.: 'fix'
  /*
  scopeOverrides: {
    fix: [
      {name: 'merge'},
      {name: 'style'},
      {name: 'e2eTest'},
      {name: 'unitTest'}
    ]
  },
  */
  // override the messages, defaults are as follows
  messages: {
    type: "Select the type of change that you're committing:",
    scope: '\nDenote the SCOPE of this change (optional):',
    // used if allowCustomScopes is true
    customScope: 'Denote the SCOPE of this change:',
    subject: 'Write a SHORT, IMPERATIVE tense description of the change:\n',
    body: 'Provide a LONGER description of the change (optional). Use "|" to break new line:\n',
    breaking: 'List any BREAKING CHANGES (optional):\n',
    footer: 'List any ISSUES CLOSED by this change (optional). E.g.: #31, #34:\n',
    confirmCommit: 'Are you sure you want to proceed with the commit above?',
  },

  allowCustomScopes: true,
  allowBreakingChanges: ['feat', 'fix'],
  // skip any questions you want
  skipQuestions: ['body'],

  // limit subject length
  subjectLimit: 100,
  // breaklineChar: '|', // It is supported for fields body and footer.
  // footerPrefix : 'ISSUES CLOSED:'
  // askForBreakingChangeFirst : true, // default is false
};
```

the ```.cz-config.js``` is in the root directory. 

Going to ```Help --> About```:

Version: 1.35.1
Commit: c7d83e57cd18f18026a8162d042843bda1bcf21f
Date: 2019-06-12T14:27:31.086Z
Electron: 3.1.8
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Linux x64 4.15.0-54-generic

**Comment by yusiwen** [comment]
My situation is even worse. It ignores `.cs-config.js` file even if a `package.json` is given.

Version: 1.35.1
Commit: c7d83e57cd18f18026a8162d042843bda1bcf21f
Date: 2019-06-12T14:30:02.622Z
Electron: 3.1.8
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.18362

**Comment by yusiwen** [comment]
I've found something. After adding `package.json`, it needs a reloading window to read `.cz-config.js`. But if `package.json` is gone, it ignores `.cz-config.js` after reloading.

**Comment by zayd62** [comment]
yeah i figured that out too. the problem is that it doesn't seem to read ```.cz-config.js``` unless there is a ```package.json``` even if ```.cz-config.js``` is in the root directory. extension relies on ```package.json``` to  find ```.cz-config.js``` otherwise it uses defaults

**Comment by zayd62** [comment]
### Problem (i think)
according to https://github.com/KnisterPeter/vscode-commitizen/blob/master/src/extension.ts, line 63 is a function ```async function readCzConfig()``` is responsible for locating a config file

https://github.com/KnisterPeter/vscode-commitizen/blob/177754933cf8315ff8fcfd84a41a1c9e905d9b34/src/extension.ts#L63-L79

putting the code through a debugger, i've noticed that if there is no ```package.json```, the function ends on line 66 so we should first check for a ```.cz-config.js``` in root directory before looking through ```package.json```

**Comment by zayd62** [comment]
created a pull request with a fix #201. if you could check it out that would be great. First time doing vscode extension development as well as typescript so there may be things wrong

**Comment by KnisterPeter** [comment]
Implemented by #203

**Comment by KnisterPeter** [comment]
Published as v0.8.3
